shared/fs/container.js

```diff
@@ -0,0 +1,16 @@
+// @flow
+import {connect, type TypedState, type Dispatch} from '../util/container'
+import * as FSGen from '../actions/fs-gen'
+import Fs from '.'
+
+const mapStateToProps = (state: TypedState) => ({
```

Comment:
Just to make sure I understand correctly, this is called whenever *any* state is updated in from Redux. Then if the object (prop) returned from this function is identical to the previous prop, it won't trigger a render. But if e.g. counter changes, it triggers a render. Does this sound correct?

Reply:
Also sort of. First it's a shallow equal compare so the entire object doesn't need to be the same. If all the properties are the same then connect will avoid the render. There are things that can subvert this but this is the common case